ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import akka.stream.ThrottleMode
+import com.daml.cliopts
+import com.daml.cliopts.Logging.LogEncoder
+import com.daml.http.dbbackend.{DbStartupMode, JdbcConfig}
+import com.daml.metrics.MetricsReporter
+import com.daml.pureconfigutils.{HttpServerConfig, LedgerApiConfig, MetricsConfig}
+import com.daml.pureconfigutils.SharedConfigReaders._
+import pureconfig.{ConfigReader, ConvertHelpers}
+import pureconfig.generic.semiauto._
+import ch.qos.logback.classic.{Level => LogLevel}
+
+import scala.concurrent.duration._
+
+private[http] object FileBasedConfig {
+  implicit val timeProviderTypeCfgReader: ConfigReader[ThrottleMode] =
+    ConfigReader.fromString[ThrottleMode](ConvertHelpers.catchReadError { s =>
+      s.toLowerCase() match {
+        case "enforcing" => ThrottleMode.Enforcing
+        case "shaping" => ThrottleMode.Shaping
+        case s =>
+          throw new IllegalArgumentException(
+            s"Value '$s' for throttle-mode is not one of 'shaping' or 'enforcing'"
```

Comment:
The error produced already reports `s` so we should not repeat it in this part of the message. Also addressed in the aforementioned commit.